Integrations/integration-Okta.yml

```diff
@@ -0,0 +1,509 @@
+                    Accept: ['application/json'],
+                    'Content-Type': ['application/json']
+                },
+                Body: readyBody
+            },
+            true
+        );
+        return parseResponse(res);
+    };
+
+
+    var get = function(args, params, uri) {
+        raw = getRaw(args, params, uri);
+        res =  parseResponse(raw);
+        if (res && res.length && res.length > 1 ) {
+            return { result: res };
+        } else {
+            return res;
+        }
+    };
+
+    var getUserId = function(args, params) {
+        username = args.username;
+        uri = '/api/v1/users?filter=profile.login+eq+"'+encodeURIComponent(username)+'"';
+        res = get(args, params, uri);
+        if (res && res.length && res.length == 1) {
+            return res[0].id;
+        } else {
+            throw 'Unable to find userId ' + username + '\n' + JSON.stringify(res);
+        }
+    };
+
+    var buildProfile = function(args) {
+        var profile = {};
+        var keys = Object.keys(args);
+        for (var i = 0 ; i < profileArgs.length; i++) {
+            if (args[profileArgs[i]]) {
+                profile[profileArgs[i]] = args[profileArgs[i]];
+            }
+        }
+        return profile;
+    };
+
+    /*
+    {
+        "password": {
+          "value": "tlpWENT2m"
+        },
+        "recovery_question": {
+          "question": "Who's a major player in the cowboy scene?",
+          "answer": "Cowboy Dan"
+        },
+        "provider": {
+          "type": "OKTA",
+          "name": "OKTA"
+        }
+    }
+    */
+    var buildCredentials = function(args) {
+      var creds = {};
+      if (args.password) {
+          var  pswd = { value: args.password };
+          creds.password = pswd;
+      }
+      if (args.passwordQuestion || args.passwordAnswer) {
+          var recovery = {
+              question: args.passwordQuestion,
+              answer: args.passwordAnswer
+          };
+          creds.recovery_question = recovery;
+      }
+      if (args.providerName || args.providerType) {
+          var provider = {
+              name: args.providerName,
+              type: args.providerType
+          };
+          creds.provider = provider;
+      }
+      return creds;
+    };
+
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            var resp = getRaw(args, params, 'api/v1/users?limit=1');
+            if (resp.StatusCode === 200) {
+                return 'ok';
+            } else if (resp.Status) {
+                return resp.Status;
+            } else {
+                return resp;
+            }
+            break;
+        case 'okta-deactivate-user':
+            var userId = getUserId(args, params);
+            var res = post(args, params, 'api/v1/users/' + userId + '/lifecycle/deactivate');
```

Comment:
Maybe it's a matter of flavour, but I prefer 
return post(args, params, 'api/v1/users/' + getUserId(args, params) + '/lifecycle/deactivate')